types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
```

Comment:
Renaming it to `title` might be confusing, since you'd then have a function and an argument both named `title`. I did a quick test and autocompletion doesn't show the argument name when implementing it like this:
```typescript
export interface Macro<T, I, E> {
	(t: T, input: I, expected: E): void;
	title? (providedTitle: string, input: I, expected: E): string;
}

const macro: Macro<...> = () => {};
macro.title = (/* cursor */) => "";

```